File: JAVA/level2/위장.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> map = new HashMap<String, Integer>();

        for(int i=0; i<clothes.length; i++){
            if(map.containsKey(clothes[i][1]))
                map.replace(clothes[i][1], map.get(clothes[i][1])+1);
            else
                 map.put(clothes[i][1], 1);
        }

        Collection<Integer> values = map.values();

        for(Integer value: values)
            answer *= (value+1);

        return answer-1;
    }
}

File: JavaScript/level2/위장.js
function solution(clothes) {
    var answer = 1;
    let cache = new Map();
    
    for(let i of clothes){
        if(cache.has(i[1]))
            cache.set(i[1], cache.get(i[1])+1);
        else
            cache.set(i[1], 1);
    }
     
    if(cache.size === 1)
        return cache.entries().next().value[1];
    
    for(let i of cache)
        answer *= (i[1]+1);
      
    return answer-1;
}